sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/storage/TokenStorageFactory.java

```diff
@@ -0,0 +1,27 @@
+package com.microsoft.appcenter.identity.storage;
+
+import android.content.Context;
+
+/**
+ * Factory class to produce instance of {@link AuthTokenStorage}.
+ */
+public class TokenStorageFactory {
+
+    /**
+     * Instance of {@link AuthTokenStorage}.
+     */
+    private static AuthTokenStorage sTokenStorageInstance;
+
+    /**
+     * Retrieves current implementation of {@link AuthTokenStorage}.
+     *
+     * @param context application context.
+     * @return instance of {@link AuthTokenStorage}.
+     */
+    public static AuthTokenStorage getTokenStorage(Context context) {
```

Comment:
missing synchronized

Reply:
should be fine if always accessing from worker thread since its storage actually, just adding annotation then